src/lib/dialog_ui.sh

```diff
@@ -241,6 +241,78 @@ function create_loading_screen_notification()
   run_dialog_command "$cmd" "$flag"
 }
 
+function spinFrame()
+{
+  local LC_CTYPE=C
+  local spin='⣾⣽⣻⢿⡿⣟⣯⣷'
+  local charwidth=3
+
+  local i=$1
+  i=$(((i + charwidth) % ${#spin}))
+  printf "%s" "${spin:$i:$charwidth}"
+}
+
+# Create simple async loading screen notification for delayed actions.
+#
+# The caller is required to use stop_loading() with the pid of this function call
+# afterwards to stop the loading screen.
+#
+# @loading_message: The message to be displayed to the user while loading.
+# @height: Menu height in lines size, the default value is 8.
+# @width: Menu width in column size, the default value is 60.
+# @flag How to display a command, the default value is
+#   "SILENT". For more options see `src/lib/kwlib.sh` function `cmd_manager`
+#
+# Return:
+# There is no return. The function just displays the infobox and returns,
+# assuming the next command is going to delay its completionand that another
+# screen is going to clear the current (the infobox).
+function create_async_loading_screen_notification()
+{
+  local loading_message="$1"
+  local height="$2"
+  local width="$3"
+  local flag="$4"
+  local cmd
+
+  flag=${flag:-'SILENT'}
+  height=${height:-'8'}
+  width=${width:-'60'}
+
+  # Escape all single quotes to avoid breaking arguments
+  loading_message=$(str_escape_single_quotes "$loading_message")
+
+  local i=0
+  while true; do
+    # Add dialog layout if there is one
+    if [[ -n "$DIALOG_LAYOUT" ]]; then
+      cmd="DIALOGRC=${DIALOG_LAYOUT} "
+    fi
+    # We should not use --clear because this flushes the infobox
+    cmd+='dialog --colors'
+    # Add Infobox screen
```

Comment:
Do we really need this inside the loop?

Reply:
We cannot remove it from the loop, because `cmd` must be reset on each iteration. If we put those lines outside of the while loop, the loop will keep appending `" --infobox $'${loading_message} ${spin}'"` to cmd without cleaning its last value.

Reply:
Oh, I see it now! Think you are right, so just ignore our comments. You can mark this conversation as resolved, also.